server/aesdsocket.c

```diff
@@ -278,17 +282,36 @@ void *thread_func( void *thread_param )
     #endif
 	// opening an existing file in write mode or creating a file with permission modes of user=rw ,group=rw ,others=rw
 	int new_tmp_fd = open( filename, O_RDWR | O_CREAT | O_APPEND, S_IWUSR | S_IRUSR | S_IWGRP | S_IRGRP | S_IROTH | S_IWOTH );
-
     if( new_tmp_fd == -1 )
     {
         syslog(LOG_ERR, "Error opening file with error %d", errno);
     }
-    int wc = write(new_tmp_fd, recieve_buffer, total_len);
-    if( wc == -1)
+    // check for IOCTL string
+    if(strncmp(recieve_buffer, ioctl_string, strlen(ioctl_string)) == 0)
     {
-        syslog(LOG_ERR, "write");
+        struct aesd_seekto seek_buffer;
+        // Reference : https://stackoverflow.com/questions/62973978/extract-multiple-integers-from-a-string-in-c
+        int count = sscanf(recieve_buffer, "AESDCHAR_IOCSEEKTO:%d,%d", &seek_buffer.write_cmd, &seek_buffer.write_cmd_offset);
+        if(count != 2)
+        {
+            syslog(LOG_ERR, "Error sscanf with error no: %d", errno);
```

Comment:
@MaanasMDK good idea to check here but you shouldn't call ioctl in this case since the seek_buffer isn't initialized.